packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -1085,6 +1125,47 @@ mixin SchedulerBinding on BindingBase {
     }
   }
 
+  final Map<Object, DartPerformanceMode> _performanceModes = <Object, DartPerformanceMode>{};
+
+  /// Request a specific [DartPerformanceMode].
+  ///
+  /// Requestor passes a handle (`requestor`) when making the request, this is typically
```

Comment:
Why do we still need to keep track of the requestor with the new handles model?

Reply:
Don't we just have to count how many handles we have given out for a given performance mode?

Reply:
Yup, I will fix this.